datafusion/datasource/src/file_scan_config.rs

```diff
@@ -239,6 +223,7 @@ pub struct FileScanConfig {
 ///     .with_limit(Some(1000))
 ///     // Project only the first column
 ///     .with_projection_indices(Some(vec![0]))
+///     .expect("Failed to push down projection")
```

Comment:
nit -- it would be nice to make example this return a real error rather than expect